Source/Core/BoogiePL.atg

```diff
@@ -520,9 +520,13 @@ Function<.out List<Declaration>/*!*/ ds.>
     if (definition != null) {
       // generate either an axiom or a function body
       if (QKeyValue.FindBoolAttribute(kv, "inline")) {
+        if (QKeyValue.FindBoolAttribute(kv, "define"))
+          SemErr("function cannot have both :inline and :define attributes");
         func.Body = definition;
       } else if (QKeyValue.FindBoolAttribute(kv, "define")) {
-        ds.Add(func.CreateFunctionDefinition(definition, kv));
+        func.DefinitionBody = func.CreateFunctionDefinition(definition);
+        if (func.TypeParameters.Count > 0)
```

Comment:
Error check before creating the function definition?

Reply:
Good point. Done.